pkg/memory-monitor/src/monitor/cgroups.c

```diff
@@ -0,0 +1,332 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <errno.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/select.h>
+#include <syslog.h>
+#include <unistd.h>
+
+#include "config.h"
+#include "event.h"
+#include "monitor.h"
+#include "util.h"
+
+#include "cgroups.h"
+
+unsigned long cgroup_get_memory_usage(const char *cgroup_name) {
+    char path[PATH_MAX + 1];
+    int fd;
+    char str_usage[256];
+    unsigned long usage_bytes;
+
+    snprintf(path, sizeof(path), "%s/%s/memory.usage_in_bytes", CGROUP_PATH_PREFIX, cgroup_name);
```

Comment:
Shouldn't this be sizeof(path)-1 but writing a NUL character to the last byte to ensure that it is always null terminited? Or has snprintf gotten smart enough to do that itself?

Reply:
As far as I know, `sprintf()` always terminates the resulting buffer with `\0`, and it's included in the size argument.

Reply:
You're  right - the snprintf() standard manpage says it does.